### PR TITLE
Removed all compilations warnings in Visual Studio project

### PR DESCRIPTION
**Pull request by cglzaguilar**
- Removed all compilations warnings in the Visual Studio project.
- Also updated the Visual Studio project to treat warnings as errors so no new warnings are introduced after this change is merged.

### CODE REVIEW

**Comment by mehagar** [comment]
Thanks so much for this! This is an investment for the future.

File: source/core/Math.cpp [hunk 1]
```diff
@@ -366,11 +364,15 @@ DECLARE_SCALE2X_INTN_HELPER(Single)
...
     DECLARE_VIREO_PRIMITIVE2(IsGE0##TYPE, TYPE, Boolean, (_Param(1) = _Param(0) >= 0)) \
+    DECLARE_VIREO_PRIMITIVE2(IsLE0##TYPE, TYPE, Boolean, (_Param(1) = _Param(0) <= 0)) \
+    DECLARE_VIREO_PRIMITIVE2(IsGT0##TYPE, TYPE, Boolean, (_Param(1) = _Param(0) >  0)) \
     DECLARE_VIREO_COMPARISON_PRIMITIVES_BASE(TYPE)
 
 #define DECLARE_VIREO_COMPARISON_PRIMITIVES_BOOLEAN() \
     DECLARE_VIREO_PRIMITIVE2(IsLT0Boolean, Boolean, Boolean, (_Param(1) = false)) \
     DECLARE_VIREO_PRIMITIVE2(IsGE0Boolean, Boolean, Boolean, (_Param(1) = true)) \
+    DECLARE_VIREO_PRIMITIVE2(IsLE0Boolean, Boolean, Boolean, (_Param(1) = (_Param(0) ? 1 : 0) <= 0)) \
+    DECLARE_VIREO_PRIMITIVE2(IsGT0Boolean, Boolean, Boolean, (_Param(1) = (_Param(0) ? 1 : 0) >  0)) \
```
Comment (sanmut) [inline source/core/Math.cpp]: These four (LT0....GT0) with boolean input appear to be semantic analysis errors in LV. So, we may not get here for LV generated VIA.

Comment (cglzaguilar) [reply source/core/Math.cpp]: I attempted to remove the Macros, but I got errors. It appears another code is expecting the functions to exists. I'll look into removing the Macros in another PRs.

File: source/core/Events.cpp [hunk 2]
```diff
@@ -194,7 +194,7 @@ class EventOracle {
         return _qObject[qID].DoneProcessingEvent();
     }
     void DeleteEventQueue(EventQueueID qID) {  // marks unallocated
-        if (qID < _qObject.size())
+        if ((size_t)qID < _qObject.size())
```
[the remark this answers is not in the document]
Comment (cglzaguilar) [reply source/core/Events.cpp]: Done, using constructor-cast instead.

**Pull request merged**